language/stdlib/transaction_scripts/doc/overview.md

```diff
@@ -1,121 +1,4001 @@
+* The <code>abort_code</code> that was raised.
+
+The <code>abort_code</code> is a <code>u64</code> that is constructed from two values:
+1. The **error category** which is encoded in the lower 8 bits of the code. Error categories are
+declared in the <code><a href="../../modules/doc/Errors.md#0x1_Errors">Errors</a></code> module and are globally unique across the Libra framework. There is a limited
+fixed set of predefined categories, and the framework is guaranteed to use these consistently.
+2. The **error reason** which is encoded in the remaining 54 bits of the code. The reason is a unique
+number relative to the module which raised the error and can be used to obtain more information about
+the error at hand. It should primarily be used for diagnosis purposes. Error reasons may change over time as the
+framework evolves.
+
+The most common set of Move abort errors that can be returned depend on the transaction script
+and they are therefore detailed in the documentation for each transaction
+script. Each abort condition is broken down into its category, reason, and a
+description of the error in the context of the particular transaction script
+e.g.,
+
+| Error Category           | Error Reason                                | Description                                               |
+| ----------------         | --------------                              | -------------                                             |
+| <code><a href="../../modules/doc/Errors.md#0x1_Errors_NOT_PUBLISHED">Errors::NOT_PUBLISHED</a></code>  | <code><a href="../../modules/doc/LibraAccount.md#0x1_LibraAccount_EPAYER_DOESNT_HOLD_CURRENCY">LibraAccount::EPAYER_DOESNT_HOLD_CURRENCY</a></code> | <code>payer</code> doesn't hold a balance in <code>Currency</code>.             |
+| <code><a href="../../modules/doc/Errors.md#0x1_Errors_LIMIT_EXCEEDED">Errors::LIMIT_EXCEEDED</a></code> | <code><a href="../../modules/doc/LibraAccount.md#0x1_LibraAccount_EINSUFFICIENT_BALANCE">LibraAccount::EINSUFFICIENT_BALANCE</a></code>       | <code>amount</code> is greater than <code>payer</code>'s balance in <code>Currency</code>. |
+
+For each of these tables, the **error categories should be considered stable**;
+any changes to these categories will be be well-publicized in advance. On the
+other hand, the **error reasons should be considered only semi-stable**; changes
+to these may occur without notice, but changes are not expected to be common.
+
+
+<a name="@Move_Explain_4"></a>
+
+#### Move Explain
+
+
+The abort conditions detailed in each transaction script are not meant to
+be complete, but the list of error categories are. For any abort conditions
+returned that are not detailed in the script's documentation the [Move
+Explain](https://github.com/libra/libra/tree/master/language/tools/move-explain)
+tool can be used to provide a human-understandable reason for the abort condition.
+Additionally, a human readable explanation should be provided in the
```

Comment:
- Might be good to say this part first, since it's likely how most users will see the explanations
- Could be useful to emphasize that `move-explain` and json-rpc will give you the same explanation. Here it sounds like you might get something different from each.